### DISCUSSION TITLE
Seamless joins?

### DISCUSSION THREAD

**Discussion by metalollie** [yermak/AudioBookConverter#421] [General]
Hi,

is there A way to make joins in chapters perfectly seamless? For spoken word audiobooks, this isn’t an issue, but I use the software to join a lot of audio dramas, and between in each chapter there is a tiny, but perceptible gap. It’s probably around half a second at most, but does interfere ever so slightly with the listening experience of full cast audios.

**Comment by yermak** [comment]
@metalollie,
In theory there should not be any gaps. Are you sure that it's not player making small pauses on new chapter?
I would recommend to try merge chapters together (right click on chapters - Combine) - this should give you 1 chapter per book and probably will give seamless experience.

**Comment by metalollie** [comment]
Hi,

Tested it using the chapter fool and removing the chapter, but there is
still a definite, albeit very short “blip” when the next track plays. I've
attached a brief sample, plus an image of the waveform, so you can see
what's happening. The paues is around the 42 sec mark on the audio.. i cut
the waveform image to make it clearer.. on there it's the small silence
aroind the 3 sec mark.

Cheers

Darren

**Comment by yermak** [comment]
Hi,
I believe I know the reason now - https://lame.sourceforge.io/tech-FAQ.txt
Not sure how easy it will be to fix by applying solution from here https://stackoverflow.com/questions/37491846/ffmpeg-gap-in-concatenated-audio-after-splitting-audio. I'm using ffmpeg within the program.

As it will require testing with different codecs... I will need to think about it.

BTW, don't see a sample or waveform image, but i think the root cause is clear.

Could you please confirm what is your source format and output format.

Thanks

**Comment by metalollie** [comment]
Hi,

I’ve attached the waveform from Audacity again below. My input files are
256k MP3 outputting to 256k M4B.
The link you gave certainly does seem to describe the problem.

Cheers

Darren

**Comment by baconcow** [comment]
I am not sure if this is related, but it seemed similar enough that I didn't want to make a separate issue.

When take multiple m4b files and merge them into a single one, the final file does not have the same number of frames as the initial files combined. This includes the individual chapter. The amount of total frames different is low (around 1.6k out of around 2.8 billion).
